### ISSUE TITLE
CoseKey key_ops setter failure

### ISSUE THREAD

**Issue by BSipos-RKF** [TimothyClaeys/pycose#49]
It would be good if the `CoseKey.key_ops` setter did some structural checking of values. I am migrating from old to new `cose` API and accidentaly left behind a non-list `key_ops` assignment which results in a strange behavior seen below.
It took some troubleshooting to figure out my problem, which could have been found easier if the setter raised an exception instead of doing what appears to be splitting a `str` value input.

```
k1 = EC2Key(
    kid=b'ExampleEC2',
    key_ops=[keyops.DeriveKeyOp],
)
print(k1)
k2 = EC2Key(
    kid=b'ExampleEC2',
    key_ops=keyops.DeriveKeyOp,
)
print(k2)
```
will result in the output:
```
<COSE_Key(EC2Key): {'KpKty': 'KtyEC2', 'KpKid': b'ExampleEC2', 'KpKeyOps': ['DeriveKeyOp']}>
<COSE_Key(EC2Key): {'KpKty': 'KtyEC2', 'KpKid': b'ExampleEC2', 'KpKeyOps': ['D', 'e', 'r', 'i', 'v', 'e', 'K', 'e', 'y', 'O', 'p']}>
```

**Comment by TimothyClaeys** [comment]
There were a bunch issues with the COSE keys in general (see issue #39). I think I fixed most of them in pull request #47 this morning. With the latest code the error you describe should be fixed.

The following code should trigger an exception _ValueError: Key operations should be a list with KeyOps values._
```
from cose.keys import EC2Key
from cose.curves import P256
from cose.keys.keyops import SignOp

ec2key = EC2Key.generate_key(curve=P256)

ec2key.key_ops = SignOp
```
While this works fine:
```
from cose.keys import EC2Key
from cose.curves import P256
from cose.keys.keyops import SignOp

ec2key = EC2Key.generate_key(curve=P256)

ec2key.key_ops = [SignOp]
print(ec2key)
<COSE_Key(EC2Key): {'EC2KpD': "b'\\x05\\xda?FN' ... (32 B)", 'EC2KpY': "b'\\x7f\\xdf\\x07\\x1f{' ... (32 B)", 'EC2KpX': "b'68\\x18\\x99\\x11' ... (32 B)", 'EC2KpCurve': 'P256', 'KpKty': 'KtyEC2', 'KpKeyOps': ['SignOp']}>

```
Also, according to the specification, each key type has some mandatory attributes. For example, for the `SymmetricKey` the mandatory attributes are KTY and K, while for EC2Key, the mandatory fields are KTY, CURVE, X, and Y (for a public key) or  
KTY, CURVE, and D (for a private key).

These constraints have also been added to the code (see [documentation](https://pycose.readthedocs.io/en/latest/cose/keys/symmetric.html))

**Comment by BSipos-RKF** [comment]
Thank you, these new behaviors will avoid the problem I ran into.
